web/modules/custom/rdf_entity/src/Entity/Query/Sparql/SparqlCondition.php

```diff
@@ -4,47 +4,785 @@
+    // default condition has to be added in order to retrieve distinct Ids.
+    if ($this->needsDefault) {
+      $this->addDefaultCondition();
+    }
+
+    // The fieldMappingConditions are added first because they are converted
+    // into a 'VALUES' clause which increases performance.
+    $this->fieldMappingConditionsToString();
+
+    // Convert the conditions.
+    $this->conditionsToString();
+
+    // Put together everything.
+    $condition_fragments = array_merge($this->tripleFragments, $this->conditionFragments);
+    $this->stringVersion = implode(self::$conjunctionMap[$this->conjunction]['delimeter'], array_unique($condition_fragments));
+    return $this->stringVersion;
+  }
+
+  /**
+   * Converts the field mapping conditions into string versions.
+   */
+  private function fieldMappingConditionsToString() {
+    foreach ($this->fieldMappingConditions as $condition) {
+      $field_name = $condition['field'] . '__' . $condition['column'];
+      $field_predicate = $this->fieldMappings[$field_name];
+      $this->addConditionFragment(self::ID_KEY . ' ' . $this->escapePredicate($field_predicate) . ' ' . $this->toVar($field_name));
+      $condition['value'] = SparqlArg::toResourceUris($condition['value']);
+      $this->addConditionFragment($this->compileValuesFilter($condition));
+    }
+  }
+
+  /**
+   * Converts the conditions into string versions.
+   */
+  private function conditionsToString() {
+    $filter_fragments = [];
+
+    foreach ($this->conditions as $condition) {
+      if ($condition['field'] instanceof ConditionInterface) {
+        $this->addConditionFragment($condition['field']->toString());
+        continue;
+      }
+
+      if ($condition['field'] === $this->idKey) {
+        $field_name = $this->fieldMappings[$condition['field']];
+        $predicate = $this->fieldMappings[$condition['field']];
+      }
+      elseif ($condition['field'] === $this->bundleKey) {
+        $this->compileBundleCondition($condition);
+        // This will be '{$bundle_key}' as it is always an 'IN' or a 'NOT IN'
+        // clause.
+        $field_name = $condition['field'];
+        $predicate = $this->fieldMappings[$condition['field']];
+      }
+      else {
+        $field_name = $condition['field'] . '__' . $condition['column'];
+        $predicate = $this->fieldMappings[$field_name];
+        $langcode = $this->getLangCode($condition['field'], $condition['column'], $condition['lang']);
+        if (in_array($condition['operator'], $this->requiresTriple) && isset($this->fieldMappings[$field_name])) {
+          $this->addConditionFragment(self::ID_KEY . ' ' . $this->escapePredicate($this->fieldMappings[$field_name]) . ' ' . $this->toVar($field_name));
+        }
+      }
+
+      $condition['value'] = $this->escapeValue($condition['field'], $condition['value'], $condition['column'], $condition['lang']);
+      $condition['field'] = $field_name;
+      switch ($condition['operator']) {
```

Comment:
The problem with this and why it is hard to split is that the sequence of converting the string is tightly attached to the code above (as a meaning). We can split it up but I think it will only make it less readable. If we are to split it, we should create a better design for the whole method.